vms/platformvm/utxo/camino_locked.go

```diff
@@ -707,6 +935,301 @@ func (h *handler) VerifyLockUTXOs(
 	return nil
 }
 
+func (h *handler) VerifyUnlockDeposit(
+	state state.Chain,
+	tx txs.UnsignedTx,
+	ins []*avax.TransferableInput,
+	outs []*avax.TransferableOutput,
+	creds []verify.Verifiable,
+	burnedAmount uint64,
+	assetID ids.ID,
+) (map[ids.ID]uint64, error) {
+	utxos := make([]*avax.UTXO, len(ins))
+	for index, input := range ins {
+		utxo, err := state.GetUTXO(input.InputID())
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to read consumed UTXO %s due to: %w",
+				&input.UTXOID,
+				err,
+			)
+		}
+		utxos[index] = utxo
+	}
+
+	return h.VerifyUnlockDepositedUTXOs(state, tx, utxos, ins, outs, creds, burnedAmount, assetID)
+}
+
+func (h *handler) VerifyUnlockDepositedUTXOs(
+	chainState state.Chain,
+	tx txs.UnsignedTx,
+	utxos []*avax.UTXO,
+	ins []*avax.TransferableInput,
+	outs []*avax.TransferableOutput,
+	creds []verify.Verifiable,
+	burnedAmount uint64,
```

Comment:
IMO this argument should be renamed from "burnedAmount" to "amountToBurn" for better understanding what's actually hapenning later in the function

Reply:
well, thats arguable, cause we'r actually verifying already sort of happend situation, where some amount assumed to be burned, thats why arg is burnedAmount, not amount to burn - we assume its already burned with this ins/outs composition